### PR TITLE
Support Dark Mode

### PR DESCRIPTION
**Pull request by cselin**
@nickoneill I did a quick spike on the status of dark mode for SwiftUI and it's looking pretty good. I made a few quick changes to get something decent, but we probably want to add official dark appearance colors for custom colors. Let me know what you think!

|About|My Impact|
|-|-|
|Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-11-03 at 13 58 07|Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-11-03 at 13 58 03|

|Reminders|Location|
|-|-|
|Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-11-03 at 13 57 58|Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-11-03 at 13 57 52|

|Detail|Script|
|-|-|
|Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-11-03 at 13 57 34|Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-11-03 at 13 57 42|

### CODE REVIEW

File: FiveCalls/FiveCalls/ScheduleReminders.swift [hunk 1]
```diff
@@ -158,8 +158,6 @@ struct DayAndTimePickers: View {
             DatePicker("", selection: $selectedTime, displayedComponents: .hourAndMinute)
                 .labelsHidden()
                 .datePickerStyle(WheelDatePickerStyle())
-                .colorInvert() // TODO: this won't work in dark mode
```
Comment (cselin) [inline FiveCalls/FiveCalls/ScheduleReminders.swift:161]: I can't seem to find a way to change the color of the date picker wheel text that's compatible with dark mode unfortunately so we might have to use defaults here.

Comment (nickoneill) [reply FiveCalls/FiveCalls/ScheduleReminders.swift:161]: 👍 it looks OK as-is

**Comment by nickoneill** [comment]
I mentioned to @heathermh the other day that I wasn't sure dark mode fits with the brand overall, but I don't dislike it after seeing the real thing. The text colors definitely need some customization for the dark background but very nice overall!

`PrimaryButton` looks fine, `fivecallsDarkBlueText` probably needs to be brighter on dark mode. We could remove the colored text on both Impact and Reminders in favor of size / weight changes to accentuate the text hierarchy.

**Comment by heathermh** [comment]
@cselin Dark mode is definitely something I'm excited to have. This is looking good so far.

**Comment by cselin** [comment]
@nickoneill I'm going to pick this up again this week, now that you've deleted the UIKit code

**Comment by cselin** [comment]
@heathermh I ended up reverting the changes for red in dark mode to match light mode as I think it looks fine as text and definitely better for the button, that is unless with think coloring the donate button red is a bit misleading as it is often used for warnings or destructive actions:

|Before|After|
|-|-|
|Simulator Screenshot - iPhone 15 - 2023-12-04 at 13 43 11|Simulator Screenshot - iPhone 15 - 2023-12-04 at 13 45 15|

@nickoneill take a look at how Your Impact looks without the colors. I think the colors actually add something here in both light and dark mode, but we can remove:

|Light Before|Light After|
|-|-|
|Simulator Screenshot - iPhone 15 - 2023-12-04 at 13 53 43|Simulator Screenshot - iPhone 15 - 2023-12-04 at 13 53 28|

|Dark Before|Dark After|
|-|-|
|Simulator Screenshot - iPhone 15 - 2023-12-04 at 13 51 03|Simulator Screenshot - iPhone 15 - 2023-12-04 at 13 53 19|

**Comment by heathermh** [comment]
> @heathermh I ended up reverting the changes for red in dark mode to match light mode as I think it looks fine as text and definitely better for the button, that is unless with think coloring the donate button red is a bit misleading as it is often used for warnings or destructive actions:
I agree that the white text on that red button looks nice and gives a little interest to that page. My worry is when we use red text, especially when it's small. I ran into that on the reminders page where we select a time and days of the week. The contrast for text that small wasn't great. We could skip using red for text, maybe make the text bigger, or highlight it some other way - maybe a red symbol or something? Open to suggestions to keep the text readable. I use this contrast calculator since it's based on upcoming accessibility guidelines: https://www.myndex.com/APCA/. It suggests that fivecallsRed is almost OK for body text (17pt default) on a white background at regular weight. It's totally unusable on black.

**Comment by nickoneill** [comment]
Agreed on changing the red for the donate button, the lighter color doesn't work on dark mode. Perhaps it deserves two variants like blue has, text color and element color?

Let's simply remove text colors on Impact for now. It's going to be hard to merge anything more significant with @heathermh's accessibility changes to that page.

**Comment by cselin** [comment]
@heathermh @nickoneill updated with your suggestions. Added new `fivecallsRedText` color to use on Reminders with acceptable lighter red for dark mode, removed text colors from Your Impact and kept single `fivecallsRed` element color for donate button in light and dark.

**Comment by heathermh** [comment]
> @heathermh @nickoneill updated with your suggestions. Added new `fivecallsRedText` color to use on Reminders with acceptable lighter red for dark mode, removed text colors from Your Impact and kept single `fivecallsRed` element color for donate button in light and dark.

I'm appreciative if somebody wants to try to improve the dark mode version of the red text color. All I did was lighten it up enough to create sufficient contrast / make it readable on black or very dark grey. It surely doesn't look very red, though.

**Review by nickoneill — APPROVED** [review]
Looks great 👍 I added a note in #331 to take a look at the launch screen in dark mode, not required to merge this though

**Comment by cselin** [comment]
Updated `fivecallsRedText` to be a bit brighter and acceptable on APCA. Decided to apply it only to reminders grid and text since it made the Donate button way too bright on the done page. If this looks good, I'll merge these changes.
Simulator Screenshot - iPhone 15 - 2023-12-13 at 11 03 21

**Comment by nickoneill** [comment]
👍 merge away

**Comment by heathermh** [comment]
@cselin As much as I'd like to keep the red text, I'm worried that the bright red on very dark grey isn't enough contrast to make it readable. If I do a color picker from the screen shots below, I get red #EA4338 and grey #1C1C1E (APCA is -34 which is only OK for spot text). How about we keep the boxes red and the text white for dark mode? I've added some other accessibility bits for the reminders, but it's on the branch I'm working on. I'm happy to merge whatever is needed once I do a PR for the bit I'm working on.

**Comment by cselin** [comment]
> @cselin As much as I'd like to keep the red text, I'm worried that the bright red on very dark grey isn't enough contrast to make it readable. If I do a color picker from the screen shots below, I get red #EA4338 and grey #1C1C1E (APCA is -34 which is only OK for spot text). How about we keep the boxes red and the text white for dark mode? I've added some other accessibility bits for the reminders, but it's on the branch I'm working on. I'm happy to merge whatever is needed once I do a PR for the bit I'm working on.

@heathermh feel free to make any additional changes to these as you work on reminders. Though wouldn't using red for the boxes have a similar issue with contrast as the text as the size of the lines is similar to the font size?

**Comment by heathermh** [comment]
> @heathermh feel free to make any additional changes to these as you work on reminders. Though wouldn't using red for the boxes have a similar issue with contrast as the text as the size of the lines is similar to the font size?

Having to read regular size red text on a dark grey background is different than seeing lines in the same colors. We have to be able to see that text more clearly in order to make out the letters. If the font was bigger, that color combo might be readable.

Understanding the letters is higher priority than seeing / kind of seeing the lines. Users who can see the red, but not quite the red letters might still be able to get some functionality from the red lines. I do have some code on the branch I'm working on to add backup / supportive functionality for the red lines. If a user is using VoiceOver, differentiate without color or reduce animation, I have an alert that tells them they need to pick a day. That way if they don't see the red or don't want the shake, they get an alert to make sure they can understand what's going on.

**Comment by cselin** [comment]
> > @heathermh feel free to make any additional changes to these as you work on reminders. Though wouldn't using red for the boxes have a similar issue with contrast as the text as the size of the lines is similar to the font size?
> 
> Having to read regular size red text on a dark grey background is different than seeing lines in the same colors. We have to be able to see that text more clearly in order to make out the letters. If the font was bigger, that color combo might be readable.
> 
> Understanding the letters is higher priority than seeing / kind of seeing the lines. Users who can see the red, but not quite the red letters might still be able to get some functionality from the red lines. I do have some code on the branch I'm working on to add backup / supportive functionality for the red lines. If a user is using VoiceOver, differentiate without color or reduce animation, I have an alert that tells them they need to pick a day. That way if they don't see the red or don't want the shake, they get an alert to make sure they can understand what's going on.

@heathermh makes sense, thanks for the details!

**Comment by nickoneill** [comment]
@heathermh if there are other changes you think we should make here, can you open an issue about it? We'll forget about this one if it's only in the comments of a closed PR

**Comment by heathermh** [comment]
@cselin you're welcome

@nickoneill I've got a few updates pending on the branch I'm working on plus I don't have a clear picture of where Main is at at this moment. After this branch I'll try to see where we're at and add issue(s) once I know if they're still there.

**Pull request merged**